xdsl/ir/core.py

```diff
@@ -971,7 +985,7 @@ class Operation(_IRNode):
     access to the operand SSAValue.
     """
 
-    results: tuple[OpResult, ...] = field(default=())
+    results: SSAValues[OpResult] = field(default=SSAValues())
```

Comment:
What do you think about adding a setter for results that can wrap an arbitrary tuple of ssa values?